### PR TITLE
[test-api-update] rename utility solids to op equivalents

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

This is a migration of a large number of tests to job/op/graph APIs. It started with an attempt to remove `execute_solid` and friends, but that ended up to be pulling on a string that had to be further unraveled to get to a passing state-- so this now includes not only `execute_solid` conversions but also a full conversion of the test-project repo and updating of associated tests.

## How I Tested These Changes

Test suite, much of which is changed.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13172?utm_source=stack-comment).

File: python_modules/libraries/dagster-dask/dagster_dask_tests/test_execute.py [hunk 1]
```diff
@@ -35,209 +27,179 @@ def simple(_):
...
+    @job(
+        executor_def=dask_executor,
+    )
+    def job_def():
+        sleepy_dask_op()
+
+    return job_def
 
 
 def test_dask_terminate():
```
Comment (smackesey) [inline python_modules/libraries/dagster-dask/dagster_dask_tests/test_execute.py:165]: Why is this test timing out?

File: examples/docs_snippets/docs_snippets_tests/intro_tutorial_tests/test_type_guide.py [hunk 2]
```diff
@@ -127,17 +132,22 @@ def double_even(even_num: EvenDagsterType) -> EvenDagsterType:
...
+        run_config={"ops": {"double_even": {"inputs": {"even_num": 2}}}},
+        do_input_mapping=False,
     ).success
 
     # Same same as above w/r/t chatting to prha
     with pytest.raises(AssertionError):
-        execute_solid(
+        wrap_op_in_graph_and_execute(
             double_even,
             run_config={"solids": {"double_even": {"inputs": {"even_num": 3}}}},
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets_tests/intro_tutorial_tests/test_type_guide.py:149]: nit: fix config?

File: python_modules/dagster/dagster/_core/execution/api.py [hunk 3]
```diff
@@ -453,7 +453,7 @@ def execute_job(
 
     `execute_job` expects a persistent :py:class:`DagsterInstance` for
     execution, meaning the `$DAGSTER_HOME` environment variable must be set.
-    It als expects a reconstructable pointer to a :py:class:`JobDefinition` so
+    It also expects a reconstructable pointer to a :py:class:`JobDefinition` so
```
Comment (dpeng817) [inline python_modules/dagster/dagster/_core/execution/api.py:456]: nice

File: examples/docs_snippets/docs_snippets_tests/intro_tutorial_tests/test_type_guide.py (hunk 2 above)
Comment (smackesey) [reply examples/docs_snippets/docs_snippets_tests/intro_tutorial_tests/test_type_guide.py:149]: just about anything you see like this is fixed upstack (this is a large stack of 20ish PRs that ends in the deletion of `PipelineDefinition`/`ModeDefinition`/`PresetDefinition`)

File: python_modules/dagster/dagster_tests/general_tests/utils_tests/test_solid_isolation.py [hunk 4]
```diff
@@ -152,103 +152,93 @@ def hello():
...
     assert result.output_value() == "hello"
-    assert result.output_values == {"result": "hello"}
-    assert result.output_values_for_solid("hello") == {"result": "hello"}
-    assert result.output_value_for_handle("hello") == "hello"
-
-    nested_result = result.result_for_node("hello")
-    assert nested_result.success
-    assert nested_result.output_value() == "hello"
-    assert len(result.node_result_list) == 1
-    assert nested_result.output_values == {"result": "hello"}
```
Comment (dpeng817) [inline python_modules/dagster/dagster_tests/general_tests/utils_tests/test_solid_isolation.py:171]: yay

File: python_modules/libraries/dagster-dask/dagster_dask_tests/test_execute.py [hunk 5]
```diff
@@ -35,209 +27,185 @@ def simple(_):
...
+
+    return job_def
 
 
+@pytest.mark.skip(
+    """
+    Fails because 'DagsterExecutionInterruptedError' is not actually raised-- there's a timeout
+    instead. It's not clear that the test ever was working-- prior to conversion to op/job/graph
+    APIs, it appears to have been mistakenly not using the dask executor. 
+    """
```
Comment (dpeng817) [inline python_modules/libraries/dagster-dask/dagster_dask_tests/test_execute.py:170]: good flag - not a blocker for this pr but should raise in eng-help and see if we can just delete.

**Review by dpeng817 — APPROVED** [review]
once tests pass lgtm

**Pull request merged**